src/util/timestamper.h

```diff
@@ -0,0 +1,92 @@
+/// \file timestamper.h
+/// \brief Emit timestamps
+/// \author Kareem Khazem <karkhaz@karkhaz.com>
+
+#ifndef CPROVER_UTIL_TIMESTAMPER_H
+#define CPROVER_UTIL_TIMESTAMPER_H
+
+#ifdef _WIN32
+#define OPT_TIMESTAMP ""
+#define HELP_TIMESTAMP ""
+#else
+#define OPT_TIMESTAMP "(timestamp):"
+
+#define HELP_TIMESTAMP                                                         \
+  " --timestamp <monotonic|wall> print microsecond-precision timestamps.\n"    \
+  "                              monotonic: timestamps are relative to an\n"   \
+  "                              arbitrary start point, guaranteed to\n"       \
+  "                              increase, formatted as s.u where 's'\n"       \
+  "                              increases every second and 'u' is a zero-\n"  \
+  "                              padded six-digit microsecond counter.\n"      \
+  "                              wall: timestamps use the format\n"            \
+  "                              %Y-%m-%dT%H:%M:%S.u where 'u' is a six-\n"    \
+  "                              digit zero-padded microsecond counter.\n"     \
+  "                              Timestamp may decrease on daylight saving\n"  \
+  "                              time changes\n"
+#endif
```

Comment:
Help messages are good.  Verbose help messages are good.  Help messages that are significantly more verbose than all of their friends ... might be better as a slightly more succinct help message and an entry in the manual.